feat: parse jsdoc tags using json5

remcohaszing opened the pull request:
This allows using non-JSON compatible JavaScript features, such as single quotes, backticks, and trailing commas. It’s similar to using `eval()`, but in a sandboxed environment.

This opens up `ts-json-schema-generator` to the following malicious input:

```ts
interface MySchema {
  /**
   * @default function() { while (true) {} }()
   */
  foo: string;
}
```

Alternatively [json5](https://github.com/json5/json5) could be used, but that introduces a new dependency.

remcohaszing committed:
feat: parse jsdoc using vm.runInNewContext

This allows using non-JSON compatible JavaScript features, such as single
quotes, backticks, and trailing commas. It’s similar to using `eval()`, but in
a sandboxed environment.

domoritz commented:
I'm not sure I want to add this. First, it means non-json is supported and we introduce the issue you describe above. Second, it means we would be further from running in a browser since we reply on node here. 

What's the use case?

remcohaszing commented:
I’m mainly interested in supporting some JavaScript structure syntax which isn’t supported by JSON.

For example tooltips in VSCode syntax highlighting for `@default` jsdoc annotations. They highlight it as JavaScript, giving the impressions JavaScript should be supported. I haven’t seen any other implementations which require it to be valid JSON.

Of course the result will be part of a JSON schema, meaning it should be JSON serializable. I.e. there’s no need to support functions or logic. It would be nice though to support single quotes and object keys without quotes at all, and extended number notations. These are all supported by `json5`.

Are you open to using `json5` instead?

domoritz commented:
I'd be okay with using json5.

remcohaszing committed:
feat: parse jsdoc tags using json5

JSON5 supports some JSON and some additional ECMAScript features, such as single
quotes, unquoted object keys, and trailing commas.

remcohaszing committed:
Merge branch 'master' into loosen-json-parsing

remcohaszing commented:
@domoritz I think there's something wrong with @vega-org-bot. It keeps updating this comment for every new release.

domoritz commented:
@hydrosquall can you look into it?
